src/liquid/components/ld-accordion/ld-accordion.tsx

```diff
@@ -0,0 +1,181 @@
+    const panel = section.querySelector('ld-accordion-panel')
+
+    const panelOffsetToScrollParent =
+      scrollParent.scrollTop +
+      panel.getBoundingClientRect().top -
+      Math.max(0, scrollParent.getBoundingClientRect().top)
+
+    const scrollPaddingTop =
+      parseFloat(window.getComputedStyle(scrollParent)['scrollPaddingTop']) || 0
+
+    // targetOffsetBottom is the distance from the scrollParent top to the
+    // bottom of the section that is being expanded in its expanded state.
+    const targetOffsetBottom =
+      panelOffsetToScrollParent +
+      Math.min(
+        panel.scrollHeight,
+        scrollParent.clientHeight - toggle.clientHeight - scrollPaddingTop
+      )
+
+    // If an accordion section expands at the bottom end of the accordion
+    // The scroll container may not have a sufficient height at that time
+    // in order to be scrollable to the target offset. In this case we
+    // postpone the scrolling to the transition end event.
+    if (
+      !this.scrollIntoViewOnTransitionEnd &&
+      scrollParent.scrollHeight < targetOffsetBottom
+    ) {
+      this.scrollIntoViewOnTransitionEnd = true
+      return
+    }
+
+    if (
+      !this.el.closest('ld-accordion-panel') &&
+      scrollParent.clientHeight + scrollParent.scrollTop < targetOffsetBottom
+    ) {
+      const prefersReducedMotion = window.matchMedia(
+        '(prefers-reduced-motion: reduce)'
+      ).matches
+
+      scrollParent.scrollTo({
+        top: targetOffsetBottom - scrollParent.clientHeight,
+        behavior: prefersReducedMotion ? 'auto' : 'smooth',
+      })
+    }
+  }
+
+  private focusToggle = (
+    currentToggle: HTMLLdAccordionToggleElement,
+    dir: 'prev' | 'next'
+  ) => {
+    const toggleToFocus = (
+      dir === 'prev'
+        ? currentToggle.parentElement.previousElementSibling
+        : currentToggle.parentElement.nextElementSibling
+    )?.querySelector('ld-accordion-toggle')
+    if (toggleToFocus) {
+      toggleToFocus.focusInner()
+    }
+  }
+
+  private onKeydown = (ev) => {
+    if (ev.target.tagName !== 'LD-ACCORDION-TOGGLE') {
+      return
+    }
+
+    switch (ev.key) {
+      case 'ArrowUp': {
+        ev.preventDefault()
+        this.focusToggle(ev.target, 'prev')
+        return
+      }
+      case 'ArrowDown': {
+        ev.preventDefault()
+        this.focusToggle(ev.target, 'next')
+        return
+      }
+    }
+  }
+
+  private onTransitionEnd = (ev) => {
+    if (
+      !this.initialized ||
+      ev.target.tagName !== 'LD-ACCORDION-PANEL' ||
+      ev.target.closest('ld-accordion') !== this.el ||
+      !ev.target.closest('ld-accordion-section').expanded ||
+      !this.scrollIntoViewOnTransitionEnd
+    ) {
+      return
+    }
+
+    this.scrollIntoView(ev.target.closest('ld-accordion-section'))
+    this.scrollIntoViewOnTransitionEnd = false
+  }
+
+  componentDidLoad() {
+    setTimeout(() => {
+      this.initialized = true
```

Comment:
I expect this to be at least an extremely rare case, if not even an impossible case that any of the above event handlers refering to `this.initialized` are being triggered before `componenDidLoad` is invoked. When is this the case?